test/parallel/test-http-invalid-urls.js

```diff
@@ -4,17 +4,15 @@ const common = require('../common');
 if (!common.hasCrypto)
   common.skip('missing crypto');
 
-const assert = require('assert');
 const http = require('http');
 const https = require('https');
-const error = 'Unable to determine the domain name';
 
 function test(host) {
   ['get', 'request'].forEach((method) => {
     [http, https].forEach((module) => {
-      assert.throws(() => module[method](host, () => {
+      common.expectsError(() => module[method](host, () => {
         throw new Error(`${module}.${method} should not connect to ${host}`);
```

Comment:
Nit: how about replacing this with `common.mustNotCall()`?

Reply:
Sure. And while I'm at it, I can add `{` and `}` around the arrow function body because there's no need for it to implicitly return a value.

Reply:
Found other problems in the process. Refactored/fixed/commented in the commit message.